ci/do_ci.sh

```diff
@@ -285,10 +285,7 @@ case $CI_TARGET in
         echo "Building API..."
         bazel build "${BAZEL_BUILD_OPTIONS[@]}" \
               -c fastbuild @envoy_api//envoy/...
-        if [[ -n "$ENVOY_API_ONLY" ]]; then
-            exit 0
-        fi
-        ;&
+        ;;
```

Comment:
i dont think we want this change - the current code is a fallthrough - if we take it out we need to ensure the current behaviour is preserved

(if only macos _could_ use a non-broken version of bash 8/)